### PR TITLE
CI workflow

### PR DESCRIPTION
**Pull request by daveads**
add automated Docker image builds for PRs

- Add API service Docker build workflow
- Add Puppeteer service Docker build workflow
- Configure Docker Hub authentication

/claim #14

### CODE REVIEW

**Comment by daveads** [comment]
Test... 

https://github.com/daveads/firecrawl-simple/actions/runs/11694973052/job/32569604382

**Comment by daveads** [comment]
Remember to add this secrets 

secrets.DOCKER_USERNAME
secrets.DOCKER_PASSWORD

**Comment by daveads** [comment]
removed pushing to docker...

**Comment by daveads** [comment]
@skeptrunedev done with this.

File: .github/workflows/build-docker-images.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./apps/api/Dockerfile
+          push: false
+          load: true
+          tags: firecrawl:pr-${{ github.event.number }}
```
Comment (cdxker) [inline .github/workflows/build-docker-images.yml:25]:
```suggestion
          tags: trieve/firecrawl:pr-${{ github.event.number }}
```

File: .github/workflows/build-docker-images.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./apps/puppeteer-service-ts/Dockerfile
+          push: false
+          load: true
+          tags: puppeteer-service-ts:pr-${{ github.event.number }}
```
Comment (cdxker) [inline .github/workflows/build-docker-images.yml:46]:
```suggestion
          tags: trieve/puppeteer-service-ts:pr-${{ github.event.number }}
```

**Review by cdxker — CHANGES_REQUESTED** [review]
Can you change the tags to be trieve/firecrawl and puppeteer-service-ts

File: .github/workflows/build-docker-images.yml (hunk 2 above)
Comment (daveads) [reply .github/workflows/build-docker-images.yml:46]: sure...

**Comment by daveads** [comment]
@cdxker  done

**Review by cdxker — APPROVED** [review]

**Pull request merged**